mirgecom/flux.py

```diff
@@ -0,0 +1,77 @@
+""":mod:`mirgecom.flux` provides inter-facial flux routines.
+
+Numerical Flux Routines
+^^^^^^^^^^^^^^^^^^^^^^^
+
+.. autofunction:: lfr_flux
+"""
+
+__copyright__ = """
+Copyright (C) 2021 University of Illinois Board of Trustees
+"""
+
+__license__ = """
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+"""
+
+
+def lfr_flux(q_tpair, compute_flux, normal, lam):
```

Comment:
It's a little bit confusing to me to see two different `flux`es, where one is a scalar and one is a vector. Would it make sense to absorb the dot product with the normal into `compute_flux`?

Reply:
I'm not sure how to do that here.  Is this true?

```
(f@normal - lam*dq) == (f - lam*dq)@normal
```